src/drt/src/pa/FlexPA_prep.cpp

```diff
@@ -180,14 +180,79 @@ void FlexPA::genAPCentered(std::map<frCoord, frAccessPointEnum>& coords,
   // If there are less than 3 coords OnGrid will create a Centered Access Point
   frCoord manu_grid = getDesign()->getTech()->getManufacturingGrid();
   frCoord coord = (low + high) / 2 / manu_grid * manu_grid;
-  auto it = coords.find(coord);
-  if (it == coords.end()) {
+
+  if (coords.find(coord) == coords.end()) {
     coords.insert(std::make_pair(coord, frAccessPointEnum::Center));
   } else {
     coords[coord] = std::min(coords[coord], frAccessPointEnum::Center);
   }
 }
 
+/**
+ * @details This follows the Tao of PAO paper cost structure.
+ * Enclosed Boundary APs satisfy via-in-pin requirement.
+ * This is the worst access point adressed in the paper
+ */
+
+void FlexPA::genAPEnclosedBoundary(std::map<frCoord, frAccessPointEnum>& coords,
+                                   const gtl::rectangle_data<frCoord>& rect,
+                                   const frLayerNum layer_num,
+                                   const bool is_curr_layer_horz)
```

Comment:
Can you move to the same place so that the changes can be seen? (after gen_initializeAccessPoints)
unless there is a reason for moving it here.

Reply:
There are no changes in `genAPEnclosedBoundary`, but it is one of the three functions that generate access points coordinates for a given type (`genAPOnTrack`, `genAPCentered`, `genAPEnclosedBoundary`). I just moved it to be closer to them instead of being far down